academy/ibc/ibc-tao-dev.md

```diff
@@ -0,0 +1,783 @@
+---
+title: "IBC/TAO"
+order: 
+description: 
+tag: deep-dive
+---
+
+# Transport, Authentication, and Ordering Layer
+
+Now that you covered the introduction and have a better understanding of how different Inter-Blockchain Communication (IBC) Protocol components and Interchain Standards (ICS) relate to each other, take a deep dive into IBC/TAO (transport, authentication, and ordering) and the [IBC module](https://github.com/cosmos/ibc-go).
+
+## Connections
+
+If you want to connect two blockchains with IBC, you will need to etablish an IBC **connection**. Connections, once established, are responsible for facilitating all cross-chain verification of IBC state.
+
+<HighlightBox type="info">
+
+The connection semantics are described in the [Interchain Standard (ICS) 3](https://github.com/cosmos/ibc/tree/master/spec/core/ics-003-connection-semantics).
+
+</HighlightBox>
+
+The opening handshake protocol allows each chain to verify the identifier used to reference the connection on the other chain, enabling modules on each chain to reason about the reference of the other chain.
+
+Take a look at the [connection Protobuf](https://github.com/cosmos/ibc-go/blob/main/proto/ibc/core/connection/v1/connection.proto) for the `ConnectionEnd`:
+
+```
+message ConnectionEnd {
+  option (gogoproto.goproto_getters) = false;
+  // client associated with this connection.
+  string client_id = 1 [(gogoproto.moretags) = "yaml:\"client_id\""];
+  // IBC version which can be utilised to determine encodings or protocols for
+  // channels or packets utilising this connection.
+  repeated Version versions = 2;
+  // current state of the connection end.
+  State state = 3;
+  // counterparty chain associated with this connection.
+  Counterparty counterparty = 4 [(gogoproto.nullable) = false];
+  // delay period that must pass before a consensus state can be used for
+  // packet-verification NOTE: delay period logic is only implemented by some
+  // clients.
+  uint64 delay_period = 5 [(gogoproto.moretags) = "yaml:\"delay_period\""];
+}
+```
+
+`ConnectionEnd` defines a stateful object on a chain connected to another separate one. Note that there must only be 2 defined `ConnectionEnd`s to establish a connection between two chains. So the connections are mapped and stored as `ConnectionEnd` on the chain.
```

Comment:
explain that connection end needs to have an associated client state, IBC is built as a stack (min 21:00 of internal recording)